bears/general/DuplicateFileBear.py

```diff
@@ -0,0 +1,32 @@
+import itertools
+
+from coalib.bears.GlobalBear import GlobalBear
+from coalib.results.Result import Result
+
+
+class DuplicateFileBear(GlobalBear):
+    LANGUAGES = {'All'}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Duplication'}
+
+    def run(self):
+        """
+        Checks for Duplicate Files using filecmp module
+        of Standard Library
+        """
+        if not self.file_dict:
+            yield Result(self, 'You did not add any file to compare')
+        else:
+            unique_tuples = [file_tuple for file_tuple in
+                             itertools.combinations(self.file_dict, 2)]
+
+            for file_pair in unique_tuples:
+                if (self.file_dict[file_pair[0]] ==
+                        self.file_dict[file_pair[1]]):
+                    first_file_name = file_pair[0]
+                    second_file_name = file_pair[1]
+                    message = ('File ' + first_file_name + ' is identical'
+                               ' to File ' + second_file_name)
```

Comment:
@meetmangukiya  Let's say you take the 0th indexed file. You can't guarantee that there are no duplicates for it unless you traverse all the 1, 2, .... n -1 files and we have to repeat the process for other n - 1 files. Total N^2 comparisions .But if you consider only unique tuples : 
```
for i in range(n-1):
    for j in range(i+1,n) : 
        Compare i'th and j'th file 
```
You reduce the number of comparisions but it is still O(N^2) (In simple terms, there exists k such that it is bounded by k* N^2 no of comparisions) . 
So , my method is optimal for any deterministic method to find duplicates :smile:  .